### PR TITLE
Tpetra:  Include multivector length in deciding where kernels should run

### PR DESCRIPTION
**Pull request by GeoffDanielson**
@trilinos/tpetra 

## Motivation
Implements @vbrunini 's changes in #7315 via Tpetra::Details::Behavior environment variables.  Defaults to his value of 10000, but allows anyone to change the threshold via the TPETRA_VECTOR_DEVICE_THRESHOLD environment variable.

## Related Issues

* Closes #7315 

## Testing
Added a couple of long multivector tests.  Existing and new unit tests pass on vesper and eclipse.

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_Details_Behavior.hpp [hunk 1]
```diff
@@ -200,6 +200,15 @@ class Behavior {
   /// separate question.
   static size_t longRowMinNumEntries ();
 
+  /// \brief the threshold for transitioning from device to host
+  ///
+  /// If the number of elements in the multivector exceeds this
```
Comment (kddevin) [inline packages/tpetra/core/src/Tpetra_Details_Behavior.hpp:205]: Should be "doesn't exceed" rather than "exceeds," right?

**Comment by vbrunini** [comment]
Is this waiting on something or can someone hit the merge button?

File: packages/tpetra/core/src/Tpetra_Details_Behavior.hpp (hunk 1 above)
Comment (brian-kelley) [reply packages/tpetra/core/src/Tpetra_Details_Behavior.hpp:205]: @vbrunini Can you please fix this comment? Then I'll approve it immediately and you can set AUTOMERGE.

Comment (vbrunini) [reply packages/tpetra/core/src/Tpetra_Details_Behavior.hpp:205]: I don't think I can push to @GeoffDanielson 's PR.

Comment (brian-kelley) [reply packages/tpetra/core/src/Tpetra_Details_Behavior.hpp:205]: @vbrunini Sorry, I thought this was your PR

**Comment by vbrunini** [comment]
Thanks @brian-kelley @GeoffDanielson !

**Pull request merged**